### PR TITLE
examples/local - vttablet-up.sh fix fore being able to run on OS X (#…

### PR DESCRIPTION
**Pull request by Rastusik**
… #1449)

### CODE REVIEW

**Comment by enisoc** [comment]
You can actually get a similar effect by not specifying the `-tablet_hostname` flag at all. Then vttablet will look up the hostname for you, and try to make sure it's fully-qualified if possible. Would you mind trying that instead? I'd prefer to let vttablet try to find the FQDN instead of using just the output of `hostname`, because the FQDN is more likely to be resolvable from other machines in the cluster.

The reason this was set to override to `hostname -i` (the IP address, rather than the hostname) is because at the time, Travis CI was assigning hostnames that were so long, MySQL would choke on them. If Travis passes with `-tablet_hostname` removed, we can go with that. If it doesn't work, maybe we can specify `hostname -i` only when the script detects that it's running in Travis.

**Comment by Rastusik** [comment]
I tried the updated vttablet-up.sh script and it work fine on OSX

**Comment by enisoc** [comment]
Thanks! Let's see if it works on Travis...

**Comment by Rastusik** [comment]
unfortunately the first build failed with this error:

Remote error: rpc error: code = 2 desc = "Tablet test-0000000101 InitSlave failed: rpc error: code = 2 desc = \"TabletManager.InitSlave on test-0000000101 error: ExecuteFetch(CHANGE MASTER TO\\n  MASTER_HOST = 'testing-worker-linux-docker-e635741f-3215-linux-5.prod.travis-ci.org',\\n  MASTER_PORT = 33100,\\n  MASTER_USER = 'vt_repl',\\n  MASTER_PASSWORD = '',\\n  MASTER_CONNECT_RETRY = 10,\\n  MASTER_USE_GTID = slave_pos) failed: String 'testing-worker-linux-docker-e635741f-3215-linux-5.prod.travis-ci.org' is too long for MASTER_HOST (should be no longer than 60) (errno 1470) during query: CHANGE MASTER TO\\n  MASTER_HOST = 'testing-worker-linux-docker-e635741f-3215-linux-5.prod.travis-ci.org',\\n  MASTER_PORT = 33100,\\n  MASTER_USER = 'vt_repl',\\n  MASTER_PASSWORD = '',\\n  MASTER_CONNECT_RETRY = 10,\\n  MASTER_USE_GTID = slave_pos\"\nTablet test-0000000102 InitSlave failed: rpc error: code = 2 desc = \"TabletManager.InitSlave on test-0000000102 error: ExecuteFetch(CHANGE MASTER TO\\n  MASTER_HOST = 'testing-worker-linux-docker-e635741f-3215-linux-5.prod.travis-ci.org',\\n  MASTER_PORT = 33100,\\n  MASTER_USER = 'vt_repl',\\n  MASTER_PASSWORD = '',\\n  MASTER_CONNECT_RETRY = 10,\\n  MASTER_USE_GTID = slave_pos) failed: String 'testing-worker-linux-docker-e635741f-3215-linux-5.prod.travis-ci.org' is too long for MASTER_HOST (should be no longer than 60) (errno 1470) during query: CHANGE MASTER TO\\n  MASTER_HOST = 'testing-worker-linux-docker-e635741f-3215-linux-5.prod.travis-ci.org',\\n  MASTER_PORT = 33100,\\n  MASTER_USER = 'vt_repl',\\n  MASTER_PASSWORD = '',\\n  MASTER_CONNECT_RETRY = 10,\\n  MASTER_USE_GTID = slave_pos\""

looks like it's the same error you were mentioning earlier.

**Comment by enisoc** [comment]
In that case, can you try using `hostname -i` if the env var [TRAVIS=true](https://docs.travis-ci.com/user/environment-variables/) and pass empty string `-tablet_hostname ''` otherwise?

**Comment by Rastusik** [comment]
This is my last try for tonight, if the build fails, I'll try tomorrow.

File: examples/local/vttablet-up.sh [hunk 1]
```diff
@@ -81,7 +86,7 @@ for uid_index in $uids; do
   $VTROOT/bin/vttablet \
     -log_dir $VTDATAROOT/tmp \
     -tablet-path $alias \
-    -tablet_hostname `hostname -i` \
+    -tablet_hostname $hostname
```
Comment (enisoc) [inline examples/local/vttablet-up.sh]: Add back the trailing ` \` and it should work. Sorry I didn't notice that.

**Comment by Rastusik** [comment]
should be fixed, my fault, sorry, it was late

**Comment by enisoc** [comment]
Thanks!

**Pull request merged**